app/views/recipes/_form.html.erb

```diff
@@ -48,6 +48,15 @@
     <%= form.check_box :is_favorite, rows: 4, class: "block shadow rounded-md border border-gray-200 outline-none px-2 py-2 mt-1" %>
   </div>
 
+   <%# <input class='form-control' list='categories' id='category_name' name='recipe[name]' value='<%= @category %> 
+  <datalist id="categories">
+        <% Category.all.each do |c| %>
+            <option> <%= c.name %> </option>
+        <% end %>
+    </datalist>
+
+   <%= form.text_field 'categories[name]', list:"categories" %>
+
```

Comment:
This will give you a basic multiselect dropdown
https://api.rubyonrails.org/classes/ActionView/Helpers/FormOptionsHelper.html
```suggestion
   <%= form.collection_select :category_ids, Category.all, :id, :name, {}, { multiple: true } %>

```

Make sure to update recipe_params in the recipes controller to include `category_ids: []`